File: leetcode/relative_sort_array/relative_sort_array.py
class Solution:
    def relativeSortArray(self, arr1: List[int], arr2: List[int]) -> List[int]:
        result = [i for i in arr1 if (i not in arr2)]
        result = sorted(result)

        for i in arr2[::-1]:
            result = [i for u in range(arr1.count(i))] + result

        return result

File: leetcode/relative_sort_array/relative_sort_array.cpp
class Solution {
public:
	vector<int> relativeSortArray(vector<int>& arr1, vector<int>& arr2) {
		map<int, int> val_frequency;
		for (int i : arr1) {
			val_frequency[i] += 1;
		}

		vector<int> ans;

		for (int i : arr2) {
			ans.insert(ans.end(), val_frequency[i], i);
			val_frequency.erase(i);
		}

		for (pair<int, int> i : val_frequency) {
			ans.insert(ans.end(), i.second, i.first);
		}

		return ans;
	}
};
